File: C++/Random.cpp
#include <iostream>
#include <ctime>

int seed;

int GenerateSeed(){
    static int x = std::time(nullptr);
    x ^= x >> 12;
    x ^= x << 25;
    x ^= x >> 27;
    return x * 0x2545F4914F6CDD1DULL;
}

int randomRange(int min, int max){
    int n = max - min + 1;
    int remainder = RAND_MAX % n;
    int x;
    do{
        x = GenerateSeed();
    } while (x >= RAND_MAX- remainder);
    
    if((min + x % n) < min){
        return std::abs((min + x % n));
    }
    else{
        return (min + x % n);
    }
}

int main() {
    seed = GenerateSeed();
    
    std::cout << randomRange(1,10) << std::endl;

    return 0;
}

File: Swift/Random.swift
import Foundation

func generateSeed() -> UInt64 {
    var x = UInt64(Date().timeIntervalSince1970)
    x ^= x >> 2;
    x ^= x << 5;
    x ^= x >> 7;
    return x * 26858
}

func randomRange(min: UInt64, max: UInt64) -> UInt64 {
    let n = max - min + 1
    let remainder = UInt64.max % n
    var x: UInt64
    repeat {
        x = generateSeed()
    } while x >= UInt64.max - remainder
    
    return min + x % n
}

func Main() {
    print(randomRange(min: 1, max: 10))
}

Main();